src/blueapi/service/main.py

```diff
@@ -177,7 +177,7 @@ def set_state(
         elif new_state == WorkerState.RUNNING:
             handler.worker.resume()
     else:
-        response.status_code = status.HTTP_400_BAD_REQUEST
+        response.status_code = status.HTTP_304_NOT_MODIFIED
```

Comment:
I'm not convinced this should change from 404. 304 is support to indicate to a client that it can continue using its cached version of a resource because the server doesn't have an updated one, i.e. it doesn't indicate that the client has requested an illegal state transition (because it's not an error code). Hence I think it should stay as BAD_REQUEST, as that's what it is.